src/commands/workspaces.ts

```diff
@@ -123,18 +120,42 @@ export class ConnectInNewWindowCommand implements Command {
 							return;
 						}
 
-						await this.initializeLocalSSH(wsData!.id);
-
 						await raceCancellationError(eventToPromise(wsState.onWorkspaceRunning), cancelToken);
+						wsData = wsState.workspaceData; // Update wsData with latest info after workspace is running
+					}
+
+					let sshDest: SSHDestination;
+					let password: string | undefined;
+					if (await this.experimentsService.getUseLocalSSHProxy()) {
```

Comment:
Should not be here a check that local ssh is initialized? i.e the logic should be similar to open URI.

Reply:
I see it is initialised above. I find it is quite confusing with so many paths. I think there should be only one function which hides the entires connectivity logic, i.e. we should not call initializeLocalSSH here and there, but only this function can. All other clients start the workspace and then call to this function. It does not need to be the part of this PR, but would be nice to clean up.